src/app/components/reportes/encabezadoReportes.component.ts

```diff
@@ -175,4 +182,22 @@ export class EncabezadoReportesComponent implements OnInit {
     add(a, b) {
         return a + b;
     }
+
+    codigoCIE10(event) {
+        if (event && event.query) {
+            const query = {
+                nombre: event.query
+            };
+            this.cie10Service.get(query).subscribe((datos) => {
+                // mapeamos para mostrar el codigo primero y luego la descripcion
+                datos.map(dato => {
```

Comment:
`map` si obtenemos un nuevo array, sino `forEach`
```suggestion
                datos.forEach(dato => {
```

Reply:
O mejor:
```
const cie10s = datos.map(dato => ({
      id: dato.id,
      nombre: `(${dato.codigo})  ${dato.nombre}`
}))
```
La explicación seria que en lugar de mandar el array con los objetos completos, retornamos unicamente las tuplas `id, nombre` que son los datos que necesitamos para desplegar las opciones

Reply:
Corregido!